projects/js-packages/ai-client/src/hooks/use-image-generator/index.ts

```diff
@@ -9,15 +9,86 @@ import requestJwt from '../../jwt/index.js';
 
 const debug = debugFactory( 'ai-client:use-image-generator' );
 
+/**
+ * Cut the post content on a given lenght so the total length of the prompt is not longer than 4000 characters.
+ * @param {string} content - the content to be truncated
+ * @param {number} currentPromptLength - the length of the prompt already in use
+ * @returns {string} a truncated version of the content respecting the prompt length limit
+ */
+const truncateContent = ( content: string, currentPromptLength: number ): string => {
+	const maxLength = 4000;
+	const remainingLength = maxLength - currentPromptLength;
+	// 6 is the length of the ellipsis and the space before it
+	return content.length > remainingLength
+		? content.substring( 0, remainingLength - 6 ) + ` [...]`
+		: content;
+};
+
+/**
+ * Create the prompt string based on the provided context.
+ * @param {string} postContent - the content of the post
+ * @param {string} userPrompt - the user prompt for the image generation, if provided. Max length is 1000 characters, will be truncated.
+ * @returns {string} the prompt string
+ */
+const getImageGenerationPrompt = ( postContent: string, userPrompt?: string ): string => {
+	/**
+	 * If the user provide some custom prompt for the image generation,
+	 * we will use it, add the post content as additional context and
+	 * provide some guardrails for the generation.
+	 */
+	if ( userPrompt ) {
+		const imageGenerationPrompt = `I need a cover image for a blog post based on this user prompt:
+
+${ userPrompt.length > 1000 ? userPrompt.substring( 0, 1000 ) : userPrompt }
+
```

Comment:
The part starting with `based on this user prompt` can be set as a separate variable that can be that text or an empty string depending on `userPrompt`. That way there is no need to repeat the whole prompt.

Reply:
Yes, for sure. I kept it this way because the prompt is still subject to changes, maybe removing rules that are not necessary when the user is in control.